packages/platform/app/controllers/profiles_controller.ts

```diff
@@ -20,13 +20,35 @@ export default class ProfilesController {
     const fitbitData = await fitbitService.getUserData(fitbitAccount);
     const fitbitUserData = fitbitData ? new FitbitUserDto(fitbitData).toJson() : null;
 
-    // NOTE: Array.from is used
-    const accounts = Array.from(user.accounts).map((account) => new AccountDto(account).toJson());
+    // Fetch devices for each account and build connected accounts array
+    const accountsWithDevices = await Promise.all(
+      Array.from(user.accounts).map(async (account) => {
+        const accountDto = new AccountDto(account).toJson();
+        let devices = null;
+
+        // Only fetch devices for Fitbit accounts for now
+        if (account.provider === 'fitbit') {
+          devices = await fitbitService.getDevices(account);
+        }
+
+        return {
+          ...accountDto,
+          devices: devices || [],
+        };
```

Comment:
```suggestion
        let devices = [];

        // Only fetch devices for Fitbit accounts for now
        if (account.provider === 'fitbit') {
          devices = await fitbitService.getDevices(account);
        }

        return {
          ...accountDto,
          devices,
        };
```

Just to avoid the `null` problem altogether.